src/modules/Product/Api/Admin.php

```diff
@@ -97,14 +97,31 @@ public function prepare($data)
             }
         }
 
+
         $types = $service->getTypes();
         if (!array_key_exists($data['type'], $types)) {
             throw new \Box_Exception('Product type :type is not registered', [':type' => $data['type']], 413);
         }
 
         $categoryId = $data['product_category_id'] ?? null;
 
-        return (int) $service->createProduct($data['title'], $data['type'], $categoryId);
+        $productID = $service->createProduct($data['title'], $data['type'], $categoryId);
+
+        // check if product type is downloadable
+        if ($data['type'] == 'downloadable') {
```

Comment:
Why is this being done here rather than from within `Servicedownloadable`?

Reply:
Because this is where the Product is created, and the product contains the config. Servicedownloadable is the instance of the downloadable file attached to the _order_ not the product., and is only created when a pdf gets uploaded. 

Technically, because I [set yes to "checked"](https://github.com/FOSSBilling/FOSSBilling/pull/1488/files#diff-46900813423ab53fe926f199e0a0eafd6ead9e42cd32ef12a7fcbbd3c771e7cdR13) if the config is empty, it would technically be defaulting to yes if the user doesn't change the setting during the upload. 

This just exists to make sure that the default value actually gets set in the database, rather than to rely on the first product upload.